external-crates/move/crates/move-package-alt/src/graph/rename_from.rs

```diff
@@ -172,4 +190,84 @@ mod tests {
         assert_snapshot!(scenario.graph_for("root").await.check_rename_from().unwrap_err().to_string(), @"TODO");
         assert_snapshot!(scenario.graph_for("a").await.check_rename_from().unwrap_err().to_string(), @"TODO");
     }
+
+    #[test(tokio::test)]
+    async fn test_modern_using_legacy_framework() {
+        let scenario = TestPackageGraph::new(["root"])
+            .add_package("std", |pkg| pkg.set_legacy().set_legacy_name("MoveStdLib"))
+            .add_package("sui", |pkg| pkg.set_legacy().set_legacy_name("Sui"))
+            .add_package("sui_system", |pkg| {
+                pkg.set_legacy().set_legacy_name("SuiSystem")
+            })
+            .add_deps([("root", "std")])
+            .add_dep("root", "sui", |dep| dep.name("my_sui").rename_from("sui"))
+            .add_dep("root", "sui_system", |dep| {
+                dep.name("my_sui_system").rename_from("sui_system")
+            })
+            .add_dep("sui", "std", |dep| dep.name("MoveStdLib"))
+            // legacy -> legacy case (SuiSystem -> MoveStdLib (std)
+            .add_dep("sui_system", "std", |dep| dep.name("MoveStdLib"))
+            .build();
+
+        scenario
+            .graph_for("root")
+            .await
+            .check_rename_from()
+            .unwrap();
+
+        scenario
+            .graph_for("sui_system")
+            .await
+            .check_rename_from()
+            .unwrap();
+    }
+
+    #[test(tokio::test)]
+    async fn test_modern_to_legacy_not_allowed_behaviours() {
+        let scenario = TestPackageGraph::new(["root", "foo", "bar", "baz"])
+            .add_package("legacy", |pkg| pkg.set_legacy().set_legacy_name("Legacy"))
+            .add_package("sui", |pkg| pkg.set_legacy().set_legacy_name("Sui"))
+            .add_package("std", |pkg| pkg.set_legacy().set_legacy_name("MoveStdLib"))
+            .add_package("malformed", |pkg| {
+                pkg.set_legacy().set_legacy_name("weird-input")
+            })
+            // 1. (FAIL) Cannot use the legacy name in the left side assignment
+            .add_dep("root", "sui", |dep| dep.name("Sui"))
```

Comment:
the name "root" confused me a bit because it's not really any different from `foo`, `bar`, and `baz`